### DISCUSSION TITLE
A code puzzle about shared AdaLN

### DISCUSSION THREAD

**Discussion by LiuHuijie6410** [PixArt-alpha/PixArt-alpha#135] [Q&A]
Thanks for such a perfect job！
I am confused about the operation of the shared AdaLN in PixArtBlock shown in the figure below, specifically the code circled in red.
Why is the shift and scale directly obtained by adding a randomly initialized parameter to t without passing it through any MLP? According to the article, shouldn't they be sent to a shared MLP?
<img width="1049" alt="image" src="https://github.com/PixArt-alpha/PixArt-alpha/assets/108620116/8d7ac875-fc8e-4df7-af11-f21f694c0b50">

**Comment by lawrence-cj** [comment]
t is the value passed through the shared MLP

**Reply by LiuHuijie6410** [reply]
But according to the picture in the paper, t should be fed to an MLP first, right?
![1714375223067](https://github.com/PixArt-alpha/PixArt-alpha/assets/108620116/bdcc901d-d425-4f1e-8f19-c885ef91ddc1)

**Reply by lawrence-cj** [reply]
U will find it in the code if reading carefully.
